spacecraft/lib/spacecraft.dart

```diff
@@ -0,0 +1,24 @@
+import 'dart:async';
+
+import 'radio.dart';
+import 'gyro.dart';
+
+class Spacecraft {
+  Radio _radio;
+  Gyro _gyro;
+
+  Spacecraft(this._radio, this._gyro);
+
+  // Acknowledge that a message is received.
+  messageReceived(String message) {
+    acknowledge();
+    _radio.increaseSignalStrength();
```

Comment:
Actually, what I meant in the example I have (copied below) is to test that when Spacecraft sends a message (through its Radio) and Radio doesn't give any reply, then the spacecraft cranks up the signal strength and tries again. 

Then we can test this behavior:

```
  var radio = new MockRadio();
  var spacecraft = new Spacecraft(radio, ...);
  when(radio.broadcast(...)).return(noReply);
  spacecraft.sendMessage("something");
  // Verify that spaceship tries to increase signal strength when its broadcast goes unackowledged.
  verify(radio.increaseSignalStrength()).called();
```

I don't think Spacecraft should automatically increase its radio's signal strength whenever it receives a message. But maybe I'm missing something.